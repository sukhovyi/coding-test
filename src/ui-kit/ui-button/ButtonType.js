import { PrimaryButton, SecondaryButton, DangerButton, SuccessButton, TertiaryButton } from './StyledButtons';

const ButtonType = Object.freeze({
  PRIMARY: 'primary',
  DANGER: 'danger',
  SUCCESS: 'success',
  SECONDARY: 'secondary',
  TERTIARY: 'tertiary',

  getButtons() {
    return {
      PrimaryButton,
      SecondaryButton,
      DangerButton,
      SuccessButton,
      TertiaryButton,
    };
  },

  getButtonByType(type) {
    const button = {
      [ButtonType.PRIMARY]: PrimaryButton,
      [ButtonType.DANGER]: DangerButton,
      [ButtonType.SUCCESS]: SuccessButton,
      [ButtonType.SECONDARY]: SecondaryButton,
      [ButtonType.TERTIARY]: TertiaryButton,
    }[type];

    if (!button) {
      throw new Error(`Button with type ${type} does not exist!`);
    }

    return button;
  },
});

export default ButtonType;
