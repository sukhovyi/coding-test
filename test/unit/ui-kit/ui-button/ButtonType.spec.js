import ButtonType from '../../../../src/ui-kit/ui-button/ButtonType';
import {
  DangerButton,
  PrimaryButton, SecondaryButton,
  SuccessButton,
  TertiaryButton,
} from '../../../../src/ui-kit/ui-button/StyledButtons';

describe('ui-kit/ui-button/ButtonType', () => {
  it('should have correct setup', () => {
    expect(ButtonType).toMatchSnapshot();
  });

  it('getButtonByType should return correct button by type', () => {
    expect(ButtonType.getButtonByType(ButtonType.DANGER)).toBe(DangerButton);
    expect(ButtonType.getButtonByType(ButtonType.SUCCESS)).toBe(SuccessButton);
    expect(ButtonType.getButtonByType(ButtonType.TERTIARY)).toBe(TertiaryButton);
    expect(ButtonType.getButtonByType(ButtonType.PRIMARY)).toBe(PrimaryButton);
    expect(ButtonType.getButtonByType(ButtonType.SECONDARY)).toBe(SecondaryButton);
  });

  it('getButtonByType should trow an error', () => {
    try {
      expect(ButtonType.getButtonByType('test-type')).toBe(PrimaryButton);
    } catch (e) {
      expect(e.message).toBe('Button with type test-type does not exist!');
    }
  });

  it('getButtons should return correct result', () => {
    expect(ButtonType.getButtons()).toEqual({
      PrimaryButton,
      SecondaryButton,
      DangerButton,
      SuccessButton,
      TertiaryButton,
    });
  });
});
