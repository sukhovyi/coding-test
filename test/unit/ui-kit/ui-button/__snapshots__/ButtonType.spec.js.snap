// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-kit/ui-button/ButtonType should have correct setup 1`] = `
Object {
  "DANGER": "danger",
  "PRIMARY": "primary",
  "SECONDARY": "secondary",
  "SUCCESS": "success",
  "TERTIARY": "tertiary",
  "getButtonByType": [Function],
  "getButtons": [Function],
}
`;
